storagenode/gc/endpoint.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/bloomfilter"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/storagenode/pieces"
+)
+
+// EndpointError defines errors class for Endpoint
+var EndpointError = errs.Class("garbage collector endpoint error")
+
+// Endpoint implements the garbage collector endpoint
+type Endpoint struct {
+	log       *zap.Logger
+	store     *pieces.Store
+	pieceinfo pieces.DB
+}
+
+// NewEndpoint creates a new endpoint
+func NewEndpoint(log *zap.Logger, store *pieces.Store, pieceinfo pieces.DB) *Endpoint {
```

Comment:
any reason to make a new endpoint struct rather than just adding this to the struct in `storagenode/piecestore/endpoint.go`?

Reply:
None that I can think of. I'm going to move it :-)